### PR TITLE
CIV-9528 Update LR Pay Immediately screen bug fix

### PR DESCRIPTION
**Pull request by kalachandrasekar1**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-9528

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RequestJudgementByAdmissionForSpecCuiCallbackHandler.java [hunk 1]
```diff
@@ -64,7 +67,9 @@ private CallbackResponse validateDefaultJudgementEligibility(CallbackParams call
         CaseData.CaseDataBuilder<?, ?> caseDataBuilder = caseData.toBuilder();
         ArrayList<String> errors = new ArrayList<>();
         if (caseData.isJudgementDateNotPermitted()) {
-            errors.add(format(NOT_VALID_DJ_BY_ADMISSION, caseData.setUpJudgementFormattedPermittedDate()));
+            LocalDate extendedRespondent1ResponseDate = deadlineCalculatorService.calculateExtendedDeadline(
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RequestJudgementByAdmissionForSpecCuiCallbackHandler.java:70]: `caseData.getRespondent1ResponseDate().toLocalDate().plusDays(5)`
Could you put that snake in separate variable please

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 2]
```diff
@@ -566,7 +568,7 @@ private String setUpPayDateToString(CaseData caseData) {
                 .format(DateTimeFormatter.ofPattern(datePattern, Locale.ENGLISH));
         }
         if (caseData.getRespondent1ResponseDate() != null) {
-            return caseData.getRespondent1ResponseDate().plusDays(5)
+            return deadlineCalculatorService.calculateExtendedDeadline(caseData.getRespondent1ResponseDate().toLocalDate().plusDays(5))
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:571]: as above

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimLipResponseFormGenerator.java [hunk 3]
```diff
@@ -43,4 +54,26 @@ public CaseDocument generate(final CaseData caseData, final String authorization
             new PDF(fileName, docmosisDocument.getBytes(), DocumentType.DEFENDANT_DEFENCE)
         );
     }
+
+    private SealedClaimLipResponseForm getRespondent1RepaymentDate(final CaseData caseData, SealedClaimLipResponseForm templateData) {
+        if (!Objects.isNull(caseData.getRespondent1ClaimResponseTypeForSpec())) {
+            if (FULL_ADMISSION.equals(caseData.getRespondent1ClaimResponseTypeForSpec()) || (PART_ADMISSION.equals(
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimLipResponseFormGenerator.java:60]:
```
FULL_ADMISSION.equals(caseData.getRespondent1ClaimResponseTypeForSpec()) || (PART_ADMISSION.equals(
                caseData.getRespondent1ClaimResponseTypeForSpec()) && caseData.getSpecDefenceAdmittedRequired() == YesOrNo.NO)
```
Into separate method =)

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimLipResponseFormGenerator.java [hunk 4]
```diff
@@ -43,4 +54,26 @@ public CaseDocument generate(final CaseData caseData, final String authorization
...
+                caseData.getRespondent1ClaimResponseTypeForSpec()) && caseData.getSpecDefenceAdmittedRequired() == YesOrNo.NO)) {
+                return addPayByDatePayImmediately(caseData, templateData);
+            }
+        }
+        return templateData;
+    }
+
+    private SealedClaimLipResponseForm addPayByDatePayImmediately(CaseData caseData, SealedClaimLipResponseForm templateData) {
+        if (caseData.isPayImmediately()) {
+            LocalDate payDeadlineDate = deadlineCalculatorService.calculateExtendedDeadline(LocalDate.now().plusDays(
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimLipResponseFormGenerator.java:70]:
```
LocalDate.now().plusDays(
                RespondentResponsePartAdmissionPaymentTimeLRspec.DAYS_TO_PAY_IMMEDIATELY)```

separate var for readability please

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java (hunk 2 above)
Comment (kalachandrasekar1) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: fixed

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimLipResponseFormGenerator.java (hunk 3 above)
Comment (kalachandrasekar1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimLipResponseFormGenerator.java]: fixed

File: src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimLipResponseFormGenerator.java (hunk 4 above)
Comment (kalachandrasekar1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sealedclaim/SealedClaimLipResponseFormGenerator.java]: updated same change

**Review by neeta-hmcts — APPROVED** [review]

**Pull request merged**